plugins/scanners/scancode/src/main/kotlin/ScanCode.kt

```diff
@@ -79,34 +81,34 @@ object ScanCodeCommand : CommandLineTool {
  *   file, the end line is set to the maximum of all end lines for per-line findings in that file, and the score is set
  *   to the arithmetic average of the scores of all per-line findings in that file.
  */
-class ScanCode internal constructor(
-    name: String,
-    private val config: ScanCodeConfig,
-    private val wrapperConfig: ScannerWrapperConfig
-) : LocalPathScannerWrapper(name) {
-    // This constructor is required by the `RequirementsCommand`.
-    constructor(name: String, wrapperConfig: ScannerWrapperConfig) : this(name, ScanCodeConfig.DEFAULT, wrapperConfig)
-
+@OrtPlugin(
+    id = "ScanCode",
+    displayName = "ScanCode",
+    description = "A wrapper for [ScanCode](https://github.com/aboutcode-org/scancode-toolkit).",
+    factory = ScannerWrapperFactory::class
+)
+class ScanCode(
+    override val descriptor: PluginDescriptor = ScanCodeFactory.descriptor,
+    private val config: ScanCodeConfig
+) : LocalPathScannerWrapper() {
     companion object {
         const val SCANNER_NAME = "ScanCode"
 
         private const val LICENSE_REFERENCES_OPTION_VERSION = "32.0.0"
         private const val OUTPUT_FORMAT_OPTION = "--json"
     }
 
-    class Factory : ScannerWrapperFactory<ScanCodeConfig>(SCANNER_NAME) {
-        override fun create(config: ScanCodeConfig, wrapperConfig: ScannerWrapperConfig) =
-            ScanCode(type, config, wrapperConfig)
-
-        override fun parseConfig(options: Options, secrets: Options) = ScanCodeConfig.create(options)
-    }
-
     private val commandLineOptions by lazy { getCommandLineOptions(version) }
 
     internal fun getCommandLineOptions(version: String) =
         buildList {
             addAll(config.commandLine)
-            addAll(config.commandLineNonConfig)
+            config.commandLineNonConfig?.let { addAll(it) }
+
+            if ("--processes" !in config.commandLineNonConfig.orEmpty()) {
+                add("--processes")
+                add(max(1, Runtime.getRuntime().availableProcessors() - 1).toString())
```

Comment:
Nit: Maybe extract the `max()` result to a variable, and just do something like `add(numProcesses.toString())` here?